src/components/Form.test.js

```diff
@@ -305,11 +330,209 @@ describe("Form", () => {
+              "address.streetNumber": "Please enter a street number",
+            });
+          }
+        }, [setErrors, state.values.name]);
+
+        return (
+          <>
+            <Input name="name" label="Name" />
+            <Input name="age" label="Age" />
+            <Input name="address.streetNumber" label="Street number" />
+            <Input name="address.streetName" label="Street name" />
+          </>
+        );
+      };
+
+      render(
+        <Form initialValues={initialValues} initialErrors={initialErrors}>
+          {RenderChild}
+        </Form>
+      );
+
+      expect(
+        screen.queryByText("This name is already taken")
+      ).not.toBeInTheDocument();
+      expect(
+        screen.queryByText("Try to spell it differently")
+      ).not.toBeInTheDocument();
+      expect(screen.queryByText("Please enter a name")).toBeInTheDocument();
+      expect(screen.queryByText("Please enter an age")).toBeInTheDocument();
+      expect(
+        screen.queryByText("Please enter a street number")
+      ).not.toBeInTheDocument();
+
+      userEvent.type(screen.getByLabelText("Name"), "Helena");
+
+      await waitFor(() => {
+        expect(
+          screen.queryByText("This name is already taken")
+        ).toBeInTheDocument();
+        expect(
+          screen.queryByText("Try to spell it differently")
+        ).toBeInTheDocument();
+        expect(screen.queryByText("Please enter a name")).not.toBeInTheDocument();
+        expect(screen.queryByText("Please enter an age")).toBeInTheDocument();
+
+        expect(
+          screen.queryByText("Please enter a street number")
+        ).toBeInTheDocument();
+      });
+    });
+
+    it("sets form state to initial values when resetForm is called without argument", async () => {
+      const initialValues = {
+        name: "",
+        address: { streetNumber: "", streetName: "" },
+      };
+      const initialErrors = {
+        name: ["Please enter a name"],
+        address: { streetNumber: ["Please enter a street number"] },
+      };
+
+      const RenderChild = ({ resetForm }) => (
+        <>
+          <Input name="name" label="Name" />
+          <Input name="address.streetNumber" label="Street number" />
+          <Input name="address.streetName" label="Street name" />
+          <Button
+            testId="resetButton"
+            onClick={() => {
+              resetForm();
+            }}
+          >
+            Reset
+          </Button>
+        </>
+      );
+
+      render(
+        <Form initialValues={initialValues} initialErrors={initialErrors}>
+          {RenderChild}
+        </Form>
+      );
+
+      expect(screen.queryByText("Please enter a name")).toBeInTheDocument();
+      expect(
+        screen.queryByText("Please enter a street number")
+      ).toBeInTheDocument();
+      expect(screen.queryByDisplayValue("Helena")).not.toBeInTheDocument();
+      expect(screen.queryByDisplayValue("22")).not.toBeInTheDocument();
```

Comment:
I suggest we remove these 4 `expect`s since they are not adding value on top of previous tests.